### PR TITLE
feat(node-rewards): introduce NodeRewardAmounts for flexible per-node reward distribution

### LINKED ISSUE #23968: (HIP-1357) Block node rewards
For HIP-1357, we'll want to:
- [ ] Ensure there are `@HapiTest` DSL "verbs" to do convenient CRUD operations for registered block nodes.
- [ ] Add `@Tag(INTEGRATION)` `@HapiTest`s that assert expected payments across the full matrix of node reward eligibility scenarios (via a reusable net-of-normal-rewards assertion pattern):
  - [ ] Inactive, no registered block node
  - [ ] Inactive, one registered block node
  - [ ] Inactive, multiple registered block nodes
  - [ ] Active, no registered block node
  - [ ] Active, one registered block node
  - [ ] Active, multiple registered block nodes
- [ ] Implement extra payments for `Node`s w/ registered block nodes.

### PR DESCRIPTION
**Pull request by viniciusjssouza**
**Description**:
Add NodeRewardAmounts value object supporting CONSENSUS_NODE and
BLOCK_NODE reward types with variable amounts. Refactor
NodeRewardManager and V0610TokenSchema to use it, deprecating
the fixed-credit overloads. 

This change prepares the system for Block Node Rewards (HIP-1357) by supporting different reward values for different nodes.

It also improves the node reward distribution system:
                                                                                                                                                                                                                                                                                  
  1. Separation of concerns — Reward calculation and reward dispatch are now decoupled. `NodeRewardManager` fully owns the business logic (who gets paid, how much, under what budget constraints), while `SystemTransactions` is reduced to pure dispatch. This makes each class  easier to understand and test in isolation.
  2. Flexible per-node reward amounts — The introduction of `NodeRewardAmounts` replaces the old model where all nodes received a single uniform `perNodeReward`. The new data structure tracks rewards per node and per reward type (consensus vs. block node), laying the groundwork for HIP-1357 block node rewards where different nodes may receive different amounts.                                                                                                                                                                                           
  3. Centralized activity and grouping logic — `NodeRewardActivity` and `NodeRewardGroups` consolidate the rules for classifying nodes as active or inactive into dedicated value types, eliminating scattered `List<AccountID>` manipulation and making the grouping logic reusable and independently testable.                                                                                                                                                                                                                                                        
  4. Improved testability — The refactoring moves reward calculation into @VisibleForTesting methods on `NodeRewardManager`, allowing unit tests to exercise budget constraint edge cases (partial inactive funding, balance-equals-activeTotal boundaries, zero-per-node rounding) without wiring up full state mocks. Test coverage grew from ~140 lines to ~420 lines while tests became more focused.                

**Related issue(s)**:

Fixes #23968 

**Notes for reviewer**:
I asked Claude to perform an initial review focusing on finding any incompatible changes with previous code. Here is what it returned:
```
 No breaking changes detected. The refactor is safe.                                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                                    
  A few specific things confirmed:                                                                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                                                                                                    
  - NodeRewardAmounts.toTransferList() produces an identical TransferList structure to the old dispatchSynthNodeRewards overloads — same credits per node, same payer debit.                                                                                                                                        
  - Old V0610TokenSchema overloads are still functional, just marked @Deprecated(forRemoval = true). No callers are broken.
  - computeActiveBlockNodeRewards() stub in NodeRewardManager is a no-op that correctly hooks into the flow without affecting existing reward computation.                                                                                                                                                          
  - Budget constraint logic (the 3 cases) is algorithmically unchanged; just restructured into calculateRewardAmounts().                                                                                                                                                                                            
  - Test coverage is thorough — all existing behavior paths, edge cases (zero amounts, pre-paid rewards, insufficient budget), and null safety are covered.                                                                                                                                                         
```

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/NodeRewardManager.java [hunk 1]
```diff
@@ -396,6 +388,227 @@ List<NodeRewardActivity> excludeNodesDecliningRewards(@NonNull final List<NodeRe
...
+            final long minNodeReward,
+            @NonNull final NodeRewardAmounts rewardAmounts) {
+        final var targetPayInTinycents = BigInteger.valueOf(nodesConfig.targetYearlyNodeRewardsUsd())
+                .multiply(USD_TO_TINYCENTS.toBigInteger())
+                .divide(BigInteger.valueOf(nodesConfig.numPeriodsToTargetUsd()));
+        final long targetNodeReward =
+                exchangeRateManager.getTinybarsFromTinycents(targetPayInTinycents.longValue(), now);
+        final long perNodeReward = Math.max(minNodeReward, targetNodeReward - prePaidRewards);
+        for (final var activity : activities) {
+            rewardAmounts.addConsensusNodeReward(activity.nodeId(), activity.accountId(), perNodeReward);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/NodeRewardManager.java:468]: I would prefer looking up block node existence and adding rewards here, instead of going through the loops of nodeActivities again in `computeActiveBlockNodeRewards`

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/NodeRewardManager.java [hunk 2]
```diff
@@ -396,6 +388,227 @@ List<NodeRewardActivity> excludeNodesDecliningRewards(@NonNull final List<NodeRe
...
+
+    /**
+     * Computes and adds consensus node rewards for inactive nodes.
+     *
+     * @param activities the inactive node activities to reward
+     * @param amount the minimum reward amount per node
+     * @param rewardAmounts the mutable reward amounts to an update
+     */
+    @VisibleForTesting
+    void computeInactiveConsensusNodeRewards(
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/NodeRewardManager.java:495]: Same here, we should loop through `nodeActivities` once and update reward amounts as needed

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! tyvm @viniciusjssouza

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/NodeRewardManager.java (hunk 1 above)
Comment (viniciusjssouza) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/NodeRewardManager.java:468]: Thanks Neeha, it makes sense! I will be implementing this change in the next PR to avoid having to sync the mirrored one
